opensearchpy/connection/krb_requests.py

```diff
@@ -0,0 +1,62 @@
+"""
+Implements a connection with Kerberos authentication support.
+Adds an Authorization header with the kerberos context during the connection phase.
+"""
+import warnings
+import kerberos
+
+from . import RequestsHttpConnection
+
+class RequestsKrbConnection(RequestsHttpConnection):
+    """ Class to connect with Kerberos """
+    def __init__(self,
+                 host="localhost",
+                 port=None,
+                 http_auth=None,
+                 use_ssl=False,
+                 verify_certs=True,
+                 ssl_show_warn=True,
+                 ca_certs=None,
+                 client_cert=None,
+                 client_key=None,
+                 headers=None,
+                 http_compress=None,
+                 opaque_id=None,
+                 **kwargs
+                 ): # pylint: disable=too-many-locals,too-many-arguments
+        """ Init class """
+        super(RequestsKrbConnection, self).__init__(
+            host=host,
+            port=port,
+            use_ssl=use_ssl,
+            verify_certs=verify_certs,
+            ssl_show_warn=ssl_show_warn,
+            ca_certs=ca_certs,
+            client_cert=client_cert,
+            client_key=client_key,
+            headers=headers,
+            http_compress=http_compress,
+            opaque_id=opaque_id,
+            **kwargs
+        )
+        self._set_kerberos_headers(host)
+
+
+    def _set_kerberos_headers(self, host):
+        """do the kerberos magic"""
+        principal = 'HTTP@' + host
+        _, _krb_context = kerberos.authGSSClientInit(principal)
+        try:
+            kerberos.authGSSClientStep(_krb_context, "")
+        except kerberos.GSSError as message:
+            warnings.warn(f"Cannot connect to the realm {principal}. More details: {message}")
```

Comment:
Showing client-side warnings isn't a pattern in opensearch-py, I couldn't find another place where this is done. I suppose `GSSError` contains enough information about the problem, just let it be? And what is the reason to wrap all exceptions into `ValueError`?